AudioDeviceExample/ViewController.swift

```diff
@@ -313,12 +416,22 @@ extension ViewController : TVIRoomDelegate {
         }
 
         self.room = nil
+        self.localAudioTrack = nil
+        self.localVideoTrack = nil
+        self.camera?.previewView .removeFromSuperview()
+        self.camera = nil;
+
         self.showRoomUI(inRoom: false)
     }
 
     func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
         logMessage(messageText: "Failed to connect to Room:\n\(error.localizedDescription)")
+
         self.room = nil
+        self.localAudioTrack = nil
+        self.localVideoTrack = nil
+        self.camera?.previewView .removeFromSuperview()
+        self.camera = nil;
```

Comment:
If we need to have the ability to teardown media (in order to switch audio devices) then can we factor out the code to do so? Perhaps a method like `unprepareMedia` would work?